### PR TITLE
Improved documentation

### PR DESCRIPTION
**Pull request by kdgutier**
The following changes are not affecting any functionality, but improving code's readability, code's documentation, and avoids duplicated `models.__init__` code.
- Improved `BaseWindows` super init: avoids duplicated `MLP-based.__init__` code, and improves `BaseWindows` methods readability.
- Improved `BaseRecurrent` super init: avoids duplicated `Recurrent-based.__init__` code, and improves `BaseRecurrent` methods readability.
- Improved/Shared order of `__init__` inputs across `MLP-based` and `RNN-based` methods.
- Docstrings documentation to `MLP`, `NBEATS`, `NHITS`, `NBEATSx`, `TFT`.
- Added `scaler_type` placeholder to `Recurrent-based` models, defaulted `normalize=True` momentarily. `normalize` argument is deprecated in favor of [scalers collection](https://nixtla.github.io/neuralforecast/common.scalers.html).

Pending:
- Docstrings documentation to `RNN`, `LSTM`, `GRU`, `DilatedRNN` and `GMMTFT`.

### CODE REVIEW

File: nbs/common.base_recurrent.ipynb [hunk 1]
```diff
@@ -50,10 +50,12 @@
     "import random\n",
```
Comment (mergenthaler) [inline nbs/common.base_recurrent.ipynb:50]: <pre class="ql-syntax" spellcheck="false">Line #33.            # TODO: clean hack and add correct use of self.scaler
</pre><p>Should we open a  issue? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/297/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (mergenthaler) [inline nbs/common.base_recurrent.ipynb:50]: <pre class="ql-syntax" spellcheck="false">Line #186.                stat_exog = windows['static'][:, static_idx]
</pre><p>Why did you remove the 0? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/297/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

File: nbs/models.nbeats.ipynb [hunk 2]
```diff
@@ -40,8 +40,7 @@
     "This method proved state-of-the-art performance on the M3, M4, and Tourism Competition datasets, improving accuracy by 3% over the `ESRNN` M4 competition winner.\n",
```
Comment (mergenthaler) [inline nbs/models.nbeats.ipynb:40]: <pre class="ql-syntax" spellcheck="false">Line #3.        """ NBEATS
</pre><p>Nbdev best practices suggests keeping docstring short and elaborate in different cels: https://nbdev.fast.ai/tutorials/best_practices.html#keep-docstrings-short-elaborate-in-separate-cells</p><p>What is our take on that? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Nixtla/neuralforecast/pull/297/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

**Comment by mergenthaler** [comment]
<p>For next time, to make the review easier,  try to make the PR shorter. For example: separate models documentation from  improvements to the BaseWindow or BaseRecurrent classes. </p><p> </p><p>It would also make it a lot easier, once that PR gets merged, to create a new issue with the pendint documentation to&nbsp;<code>RNN</code>,&nbsp;<code>LSTM</code>,&nbsp;<code>GRU</code>,&nbsp;<code>DilatedRNN</code>&nbsp;and&nbsp;<code>GMMTFT</code>. </p>

File: nbs/common.base_recurrent.ipynb (hunk 1 above)
Comment (kdgutier) [reply nbs/common.base_recurrent.ipynb:50]: This is the issue of the pending inclusion of the `scaler_type` input to RNN-based models
https://github.com/Nixtla/neuralforecast/issues/295

Comment (kdgutier) [reply nbs/common.base_recurrent.ipynb:50]: I simplified the wrangling of static variables, and passed forward to the models the responsibility to repeat the statics across the temporal dimension. 

This makes more efficient the `TimeSeriesLoader` and improves its memory usage.
As it is now moving statics of shape `[N,S_in]` rather than shape `[N,T,S_in]`.

The usage on `MLP-based` models is still on the work.

File: nbs/models.nbeats.ipynb (hunk 2 above)
Comment (kdgutier) [reply nbs/models.nbeats.ipynb:40]: We are not using documentation for the classes and methods as created by `nbdev`, because it leaves the code without docstrings.

We are waiting for `nbdev` to allow for the automatic generation of docstrings, which might simplify things in the future.

**Review by mergenthaler — APPROVED** [review]
